code/_helpers/icons.dm

```diff
@@ -911,3 +911,35 @@ proc/generate_image(var/tx as num, var/ty as num, var/tz as num, var/range as nu
 
 	return cap
 
+
```

Comment:
I moved this out of table code and into its own helper proc, it's a very useful thing to have